### PR TITLE
Added translate button to header actions dropdown along with some cleanup

### LINKED ISSUE #8866: Page listing header: clean up button hooks
Follow-up to #8792, to further change the server-side implementation of our three configurable button groups.

## is_parent

Our button hooks currently have an `is_parent` boolean flag, so users can decide whether the buttons should be added or not depending on whether the hook is running for the "header" / "parent" page, or for a "row" / "child" page in a listing.

We now use completely separate hooks for those use cases, so this attribute can probably be either removed or at least deprecated. This needs to happen for:

- register_page_listing_buttons hook
- register_page_listing_more_buttons hook
- ButtonWithDropdownFromHook, which could be reused separately (for example in the docs there is a `my_button_dropdown_hook` hook)
- register_page_header_buttons – this already isn’t supporting `is_parent`, but we nonetheless have it defined [in our usage of the hook](https://github.com/wagtail/wagtail/blob/main/wagtail/admin/wagtail_hooks.py#L366)

## "Translate this page" action

This is one action we haven’t moved from the "More" menu to the Actions dropdown in the header. I think it would make sense to have it there _in addition_ to keeping it under "More" in listings. (see also #7260)

https://github.com/wagtail/wagtail/blob/main/wagtail/contrib/simple_translation/wagtail_hooks.py#L51

### PR DESCRIPTION
**Pull request by PaarthAgarwal**
Fixes #8866.
I have added translate button to the header actions dropdown.
Removed all the instances of is_parent except those in docs. I didn't observe it breaking anything but I think it should be tested more thoroughly.
After:
Steps to reproduce:
1. Go to wagtail admin.
2. In the slim sidebar, go to settings.
3. Add a new locale.
4. Go to page listings and click on the actions dropdown button.
Screenshot from 2022-07-23 16-07-43

### CODE REVIEW

File: wagtail/admin/wagtail_hooks.py [hunk 1]
```diff
@@ -460,6 +447,29 @@ def page_header_buttons(page, page_perms, is_parent=False, next_url=None):
             },
             priority=70,
         )
+    if (
+        page_perms.user.has_perm("simple_translation.submit_translation")
+        and not page.is_root()
+    ):
```
Comment (lb-) [inline wagtail/admin/wagtail_hooks.py:453]: maybe run the tests with this commented out temporally (locally) - most likely this is the cause of the extra query and that should be ok.

But be sure that you are confident the extra query is not an accident.


```
FAIL: test_page_edit_num_queries (wagtail.admin.tests.pages.test_edit_page.TestPageEdit)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/circleci/project/wagtail/admin/tests/pages/test_edit_page.py", line 1282, in test_page_edit_num_queries
    self.client.get(
  File "/home/circleci/project/.venv/lib/python3.8/site-packages/django/test/testcases.py", line 100, in __exit__
    self.test_case.assertEqual(
AssertionError: 39 != 38 : 39 queries executed, 38 expected
```

**Comment by lb-** [comment]
@thibaudcolas  any chance you could take a look at this one, otherwise I will try to get my head around it.

File: wagtail/admin/wagtail_hooks.py (hunk 1 above)
Comment (zerolab) [reply wagtail/admin/wagtail_hooks.py:453]: This should ideally check if the `simple_translation` app is enabled, then perform the permission check.

There's an argument about better caching permission checks on the request, but that's a different story

**Comment by PaarthAgarwal** [comment]
Screenshot from 2022-07-26 15-33-32
How about one more extra query hahaha

**Comment by lb-** [comment]
Commit the change to the test so it all passes - I will take a look at it but I think that will be ok.

File: wagtail/admin/wagtail_hooks.py (hunk 1 above)
Comment (lb-) [reply wagtail/admin/wagtail_hooks.py:452]: note: the simple translation permissions will only exist IF the app is installed.

**Review by lb- — APPROVED** [review]
changes work well - but just a heads up - the documentation needed updating but I have sorted that (mostly find & remove).

**Pull request merged**